helper/trace/debug_trace.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package trace
+
+import (
+	"fmt"
+	"os"
+	"runtime/trace"
+	"time"
+)
+
+func StartDebugTrace(filePrefix string) (file string, stop func() error, err error) {
+	path := fmt.Sprintf("%s/%s_%s", os.TempDir(), filePrefix, time.Now().Format(time.RFC3339))
+	traceFile, err := os.Create(path)
```

Comment:
still want to make this more similar to `VAULT_STACKTRACE_WRITE_TO_FILE` for the sake of consistency